### PR TITLE
Implementation and documentation of AdvertServe <amp-ad> tag

### LINKED ISSUE #5364: Intent to implement: `amp-ad` type for AdvertServe
Hi,

I intend to implement an `amp-ad` type for @AdvertServe (http://www.advertserve.com)

Actually, I have already completed the implementation and signed a CLA for our parent company Renegade Internet, Inc so I will send a pull request along shortly.

I did also implement hooks into the AMP ad viewability and resizing APIs on our end.  Our "code wizard" can also generate the `amp-ad` tags for our clients to simplify things.  I was pleased to see the viewability API works like IntersectionObserver as that made it quite simple to plug into.

Thanks,
Mike

### PR DESCRIPTION
**Pull request by renegademike**
Here is my implementation and documentation of the `amp-ad` tag for @AdvertServe as indicated in #5364 

Fixes #5364.

### CODE REVIEW

File: examples/ads.amp.html [hunk 1]
```diff
@@ -277,6 +278,15 @@ <h2 id="aduptech">Ad Up Technology</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="advertserve">AdvertServe</h2>
+  <amp-ad width=300 height=250
+          type="advertserve"
+          data-client="tester"
+          data-pid=0
+          data-zid=68>
+    <div placeholder></div>
```
Comment (lannka) [inline examples/ads.amp.html]: please add the `<fallback>` tag as well.

File: examples/ads-legacy.amp.html [hunk 2]
```diff
@@ -125,6 +125,14 @@
       data-adtest="1">
   </amp-ad>
 
+  <h2>AdvertServe</h2>
```
Comment (lannka) [inline examples/ads-legacy.amp.html]: no change needed in this file

**Comment by renegademike** [comment]
Ok, I've removed my changes to examples/ads-legacy.amp.html and added the <fallback> tag as requested.

I have also implemented the renderStart, noContentAvailable and reportRenderedEntityIdentifier APIs.

**Comment by renegademike** [comment]
Thanks, I've resolved the merge conflicts.

**Pull request merged**